src/dotnet/APIView/APIViewWeb/Managers/CommentsManager.cs

```diff
@@ -148,6 +165,98 @@ public async Task<CommentItemModel> UpdateCommentAsync(ClaimsPrincipal user, str
             return comment;
         }
 
+        public async Task RequestAgentReply(ClaimsPrincipal user, CommentItemModel comment, string activeRevisionId)
+        {
+            string language = await _reviewRepository.GetLanguageReview(comment.ReviewId);
+            var commentResult = new CommentItemModel
+            {
+                ReviewId = comment.ReviewId,
+                APIRevisionId = comment.APIRevisionId,
+                SampleRevisionId = comment.SampleRevisionId,
+                ElementId = comment.ElementId,
+                ResolutionLocked = false,
+                CreatedBy = ApiViewConstants.BotName,
+                CreatedOn = DateTime.UtcNow,
+                CommentType = CommentType.SampleRevision
+            };
+
+            if (!IsUserAllowedToChatWithAgent(user, language))
+            {
+                commentResult.CommentText =
+                    "Interaction with the agent is restricted. Only designated language architects are authorized to use this feature.";
+                await _commentsRepository.UpsertCommentAsync(commentResult);
```

Comment:
This will leave unnecessary trail of comments in the review if non-architects started creating a comment. Can you add a check at the beginning and fail the request with if user does not have permission to tag agent.